boa/src/syntax/lexer/number.rs

```diff
@@ -0,0 +1,334 @@
+                    match cursor.peek()? {
+                        Some('e') | Some('E') => {
+                            // Consume the ExponentIndicator.
+                            cursor.next();
+
+                            buf.push('E');
+
+                            take_signed_integer(&mut buf, cursor, &kind)?;
+                        }
+                        Some(_) | None => {
+                            // Finished lexing.
+                        }
+                    }
+                }
+            }
+            Some('e') | Some('E') => {
+                kind = NumericKind::Rational;
+                cursor.next(); // Consume the ExponentIndicator.
+                buf.push('E');
+                take_signed_integer(&mut buf, cursor, &kind)?;
+            }
+            Some(_) | None => {
+                // Indicates lexing finished.
+            }
+        }
+
+        check_after_numeric_literal(cursor)?;
+
+        let num = match kind {
+            NumericKind::BigInt(base) => {
+                Numeric::BigInt(
+                    BigInt::from_string_radix(&buf, base as u32).expect("Could not convert to BigInt")
+                    )
+            }
+            NumericKind::Rational /* base: 10 */ => {
+                let val = f64::from_str(&buf).expect("Failed to parse float after checks");
+                let int_val = val as i32;
+                if (int_val as f64) == val {
+                    Numeric::Integer(int_val)
+                } else {
+                    Numeric::Rational(val)
+                }
+            },
+            NumericKind::Integer(base) => {
+                if let Ok(num) = i32::from_str_radix(&buf, base as u32) {
+                    Numeric::Integer(num)
+
+                    // if exp_str.is_empty() {
+                        
+                    // } else {
+                    //     let n = i32::from_str(&exp_str).map_err(|_| Error::syntax("Could not convert value to f64"))?;
+
+                    //     if n < 0 { // A negative exponent is expected to produce a decimal value.
+                    //         Numeric::Rational(
+                    //             (num as f64) * f64::powi(10.0, n)
+                    //         )
+                    //     } else if let Some(exp) = i32::checked_pow(10, n as u32) {
+                    //            if let Some(val) = i32::checked_mul(num, exp) {
+                    //                Numeric::Integer(val)
+                    //            } else {
+                    //                 Numeric::Rational(
+                    //                     (num as f64) * (exp as f64)
+                    //                 )
+                    //            }
+                    //     } else {
+                    //         Numeric::Rational(
+                    //             (num as f64) * f64::powi(10.0, n)
+                    //         )
+                    //     }
+                    // }
```

Comment:
```suggestion
```
we don't need these comments, right?

Reply:
Yes, this + handling the errors is coming in the next commit (just finishing it off now)